### PR TITLE
Remove pydantic-settings as a dependency

### PR DESCRIPTION
**Pull request by Devansh3712**
Use `dotenv` library's `load_dotenv()` function instead of `pydantic-settings` for loading environment variables from `.env` file

## Checklist
- [x] Changes tested in [container setup](./../README.md#option-1-teuthology-docker-setup)
- [x] Changes tested in [non-containerized setup](./../README.md#option-2-non-containerized-with-venv-and-pip)
- [ ] Includes tests
- [ ] Documentation updated

### CODE REVIEW

**Comment by kamoltat** [comment]
CI failure
```
#10 9.122 ERROR: Package 'teuthology-1.1.1.dev729-g861a8dcf' requires a different Python: 3.8.10 not in '>=3.10'
#10 ERROR: process "/bin/sh -c pip3 install -e ." did not complete successfully: exit code: 1
------
 > [6/7] RUN pip3 install -e .:
4.776 Collecting teuthology@ git+https://github.com/ceph/teuthology#egg=teuthology[test]
4.776   Cloning https://github.com/ceph/teuthology to /tmp/pip-install-g6d3wo0z/teuthology
4.776   Running command git clone -q https://github.com/ceph/teuthology /tmp/pip-install-g6d3wo0z/teuthology
6.291   Installing build dependencies: started
8.420   Installing build dependencies: finished with status 'done'
8.421   Getting requirements to build wheel: started
8.735   Getting requirements to build wheel: finished with status 'done'
8.739     Preparing wheel metadata: started
9.044     Preparing wheel metadata: finished with status 'done'
9.122 ERROR: Package 'teuthology-1.1.1.dev729-g861a8dcf' requires a different Python: 3.8.10 not in '>=3.10'
------
Dockerfile:24
--------------------
  22 |     WORKDIR /teuthology_api
  23 |     COPY . /teuthology_api/
  24 | >>> RUN pip3 install -e .
  25 |     RUN mkdir /archive_dir/
  26 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c pip3 install -e ." did not complete successfully: exit code: 1
Service 'teuthology_api' failed to build : Build failed
Error: Process completed with exit code 1.
```

File: src/teuthology_api/config.py [hunk 1]
```diff
@@ -1,43 +0,0 @@
...
-    session_secret_key: str = ""
-    archive_dir: str = ""
-    teuthology_path: str = ""
-
-    model_config = SettingsConfigDict(
-        env_file=".env", env_file_encoding="utf-8", extra="ignore"
-    )
-    # TODO: team names need to be changed below when created
-    admin_team: str = "ceph"  # ceph's github team with *sudo* access to sepia
-    teuth_team: str = "teuth"  # ceph's github team with access to sepia
```
Comment (kamoltat) [inline src/teuthology_api/config.py:32]: @VallariAg should we be worried about `admin_team` and `teuth_team` variables? I know you had the admin privilege change PR, which is an auxiliary to your kill PR.

**Review by VallariAg — APPROVED** [review]
Looks good to me! Maybe we can add a comment in the commit message about why we are changing this just to document it somewhere. 
From what I understand, we are doing this because we are not using any complex settings types which pydantic-settings provides, so we are switching to reading env variables using dotenv and removing an extra dependency.

**Comment by Devansh3712** [comment]
sure, i'll amend the commit message

**Pull request merged**